lib/srv/alpnproxy/aws_local_proxy.go

```diff
@@ -47,8 +47,14 @@ func (m *AWSAccessMiddleware) CheckAndSetDefaults() error {
 }
 
 func (m *AWSAccessMiddleware) HandleRequest(rw http.ResponseWriter, req *http.Request) bool {
-	if err := aws.VerifyAWSSignature(req, m.AWSCredentials); err != nil {
-		m.Log.WithError(err).Error("AWS signature verification failed.")
+	// It's found that some AWS clients may have slightly different signing
+	// logic than the Go SDK. For example, the Java SDK v1 includes the
+	// "Use-Agent" header when signing but the Go SDK explicitly excludes this
+	// header. To workaround, only AccessKeyID from the signature is verified
+	// here. A new UUID should be generated as the AccessKeyID per tsh command
+	// for better security.
+	if err := aws.VerifyAWSAccessKeyID(req, m.AWSCredentials); err != nil {
```

Comment:
If we drop signature validation, does that mean that the AccessKeyID is the only data one needs to get past this check? Isn't this weakening the properties of the system?